core-bundle/src/Resources/contao/templates/_new/content_element/code.html.twig

```diff
@@ -0,0 +1,34 @@
+{% extends "@Contao/content_element/_base.html.twig" %}
+
+{% set highlighter_css_file = 'vendor/scrivo/highlight_php/styles/foundation.css' %}
+
+{% block before %}
+    {{ parent() }}
+    {% block style %}
+        {% add 'highlighter_css' to head %}
+            {# Lazy load styles, see https://web.dev/defer-non-critical-css/ #}
+            <link rel="preload" href="{{ highlighter_css_file }}" as="style" onload="this.onload=null;this.rel='stylesheet'">
+            <noscript>
+                <link rel="stylesheet" href="{{ highlighter_css_file }}">
+            </noscript>
+        {% endadd %}
+    {% endblock %}
+{% endblock %}
+
+{% block content %}
+    <pre>
+        {%- apply spaceless %}
```

Comment:
Doesn’t this remove whitespace inside the `<pre>` too?
Or do `apply` blocks stop on a `block` boundary?

Reply:
The `spaceless` filter is supposed to remove whitespaces between HTML tags. I thought it respects `code` tags, but it turns out it's implemented quite naively: 

```php
return trim(preg_replace('/>\s+</', '><', $content ?? ''));
```

We should probably use individual whitespace control here. I'll check…

Reply:
see 48690c16cfe0877cc6c1e65e0fbca270e7acf006 (includes some more refactorings for the code template)